examples/stable-diffusion/training/train_controlnet.py

```diff
@@ -0,0 +1,1176 @@
+#!/usr/bin/env python
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+
```

Comment:
Also, can you add where this code is referenced from? Something similar like https://github.com/huggingface/optimum-habana/blob/main/examples/stable-diffusion/training/train_text_to_image_sdxl.py#L17

Reply:
updated